inappmessaging/src/main/java/com/rakuten/tech/mobile/inappmessaging/runtime/workmanager/workers/MessageEventReconciliationWorker.kt

```diff
@@ -45,7 +47,11 @@ internal class MessageEventReconciliationWorker(
         }
 
         // Messages list shouldn't be empty, if it is, then there's no more work to be done.
-        val messageListCopy = pingRepo.getAllMessagesCopy()
+        val messageListCopy = pingRepo.getAllMessagesCopy().filter {
+            // remove outdated messages
+            it.hasNoEndDate() ||
+            it.getMessagePayload().messageSettings.displaySettings.endTimeMillis > Date().time
```

Comment:
maybe `>=` is better

Reply:
Hmm, I don't think so, normally the endTime is included in the allowed time.

Reply:
you mean not included?

Reply:
I was confused by my own comment 🥴, the filter keeps only true statements. Updated the code.